File: java-boot/src/test/java/se/callistaenterprise/kotlindsl/repo/ProductRepositoryTest.java
package se.callistaenterprise.kotlindsl.repo;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import se.callistaenterprise.kotlindsl.model.ProductIn;
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest;
import se.callistaenterprise.kotlindsl.testutil.DslJooqTest;

import static org.junit.jupiter.api.Assertions.assertEquals;

@DslJooqTest
public class ProductRepositoryTest extends DslIntegrationTest {

    private final ProductRepository productRepo;

    public ProductRepositoryTest(@Autowired ProductRepository productRepo) {
        this.productRepo = productRepo;
    }

    @Test
    public void roundTripWorks() {
        var productIn = new ProductIn("1234", "A product");
        var id = productRepo.create(productIn);
        var product = productRepo.getById(id);
        assertEquals(product.productId(), productIn.productId());
        assertEquals(product.productName(), productIn.productName());
    }

}

File: kotlin-boot/src/test/kotlin/se/callistaenterprise/kotlindsl/repo/ProductRepositoryTest.kt
package se.callistaenterprise.kotlindsl.repo

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import se.callistaenterprise.kotlindsl.model.ProductIn
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest
import se.callistaenterprise.kotlindsl.testutil.DslJooqTest

@DslJooqTest
class ProductRepositoryTest(@Autowired val productRepo: ProductRepository) : DslIntegrationTest() {

    @Test
    fun `Roundtrip works`() {
        val productIn = ProductIn("1234", "A product")
        val id = productRepo.create(productIn)
        val product = productRepo.getById(id)!!
        product.productId shouldBe productIn.productId
        product.productName shouldBe productIn.productName
    }

}
